client/src/components/PlanForm.jsx

```diff
@@ -0,0 +1,287 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useUser } from '../context/UserContext';
+import { Button, Form, Input, Radio, Popover, Checkbox, Select, Modal } from 'antd';
+import { useNavigate } from 'react-router-dom'
+import GeneratedPlan from './GeneratedPlan';
+
+const PlanForm = () => {
+  const [usingDonor, setUsingDonor] = useState(false);
+  const [knownFertilityIssues, setKnownFertilityIssues] = useState(false);
+  const [generatedPlan, setGeneratedPlan] = useState(null);
+  const { user, setUser } = useUser();
+  const navigate = useNavigate();
+  const hasWarnedRef = useRef(false);
+
+  //user must be logged in to generate plan
+  useEffect(() => {
+    if (!user && !hasWarnedRef.current) {
+      hasWarnedRef.current = true; 
+      // Show warning modal
+      Modal.warning({
+        title: 'Login Required',
+        content: 'Please sign up or login to generate your conception plan.',
+        onOk: () => {
+          navigate('/');
+        },
+      });
+    }
+  }, [user, navigate]);
+ 
+ 
+
+  //<---------------------External API request-------------------------->//
+  const handleGeneratePlan = async (values) => {
+    console.log('Form values:', values);
+    //use info from user profile 
+    const formData = {
+      ...values,
+      userId: user.user_id,
+      userAge: user.age,
+      location: user.location,
+      userFamilyStructure: user.family_structure,
+      userPartner: user.has_partner,
+      userPartnerAge: user.partner_age
+    };
+    console.log("Form data", formData)
+    try {
+      const response = await fetch(`http://localhost:8080/plans/generate/${user.id}`, {
```

Comment:
You could use some environment variables to dynamically build this URL based on the environment (local, preview, testing, production). This can help streamline deployments without having to implement code changes.

Reply:
Added variables for fetch requests made on client side. [346b327](https://github.com/MBHunter88/queer_conceptions/pull/5/commits/346b32758862b85d49aaf85ebe36f96010e36584)